Content/Items/Currency/GlimmeringShard.cs

```diff
@@ -25,7 +25,13 @@ public override void SetDefaults()
 
 	public override bool CanRightClick()
 	{
-		return Main.LocalPlayer.HeldItem.TryGetGlobalItem(out PoTGlobalItem _) && Main.LocalPlayer.HeldItem.GetInstanceData().Rarity == ItemRarity.Magic;
+		if (!Main.LocalPlayer.HeldItem.TryGetGlobalItem(out PoTGlobalItem _))
+		{
+			return false;
+		}
+
+		PoTInstanceItemData data = Main.LocalPlayer.HeldItem.GetInstanceData();
+		return data.Rarity == ItemRarity.Magic && !data.Corrupted;
```

Comment:
It might be worth it to have a CanRightClick in CurrencyShard and do the
```c#
		if (!Main.LocalPlayer.HeldItem.TryGetGlobalItem(out PoTGlobalItem _))
		{
			return false;
		}
		if (data.Corrupted) return false;
```

In it so we can call `super.CanRightClick()` first or something no?

Reply:
That way we aren't needing to duplicate this a ton

Reply:
Ahh, I see this exists in the CanRightClick in CurrencyShard. Are we not able to consolidate the logic so every single CurrencyShard needs to check it's own corrupted & TryGetGlobalItem?

Reply:
Fixed.